synapse_domain_rule_checker/__init__.py

```diff
@@ -172,15 +183,51 @@ async def _user_may_invite(
 
         if (
             published_room
-            and invitee_domain
-            in self._domains_prevented_from_being_invited_to_published_rooms
+            and self.match_from_domains(self._domains_prevented_from_being_invited_to_published_rooms, 
+                                        invitee_domain)
         ):
             return False
 
-        if inviter_domain not in self._domain_mapping:
+        if not self.is_domain_mapping_passing(inviter_domain, invitee_domain):
+            return False
+
+        return True
+
+    def is_domain_mapping_passing(self, inviter_domain: str, invitee_domain: str) -> bool:
+        matching_invitees = self.domain_matching_invitees(inviter_domain) 
+        if matching_invitees is None:
             return self._config.can_invite_if_not_in_domain_mapping
+        
+        return self.match_from_domains(matching_invitees, invitee_domain)
+
+
+    def domain_matching_invitees(self, inviter_domain: str) -> Optional[set[Pattern | str]]:
+        if not self._config.use_regex:
+            return self._domain_mapping.get(inviter_domain)
+
+        matched = False
+        matching = []
+
+        for inviter, invitees in self._domain_mapping.items():
+            if inviter.fullmatch(inviter_domain):
+                matched = True
+                print(invitees)
```

Comment:
Why are you printing here?

Reply:
Smh I searched for loose debug prints, how did I miss one?